pkg/cmd/dev/ui.go

```diff
@@ -248,3 +247,205 @@ Replaces 'make ui-lint'.`,
 
 	return lintCmd
 }
+
+// arrangeFilesForTestWatchers moves files from Bazel's build output directory into the locations they'd be found during
+// a non-Bazel build, so that test watchers can successfully operate outside of the bazel sandbox.
+func arrangeFilesForTestWatchers(d *dev) error {
+	bazelBin, err := d.getBazelBin(d.cli.Context())
+	if err != nil {
+		return err
+	}
+
+	ossProtobufSrc := filepath.Join(bazelBin, "pkg", "ui", "workspaces", "db-console")
+	cclProtobufSrc := filepath.Join(ossProtobufSrc, "ccl")
+
+	dstDirs, err := getUIDirs(d)
+	if err != nil {
+		return err
+	}
+	ossProtobufDst := dstDirs.dbConsole
+	cclProtobufDst := filepath.Join(ossProtobufDst, "ccl")
+
+	protoFiles := []string{
+		filepath.Join("src", "js", "protos.js"),
+		filepath.Join("src", "js", "protos.d.ts"),
+	}
+
+	// Delete and recreate protobuf client files that were previously copied out of the sandbox
+	for _, relPath := range protoFiles {
+		ossDst := filepath.Join(ossProtobufDst, relPath)
+		cclDst := filepath.Join(cclProtobufDst, relPath)
+		if err := d.os.Remove(ossDst); err != nil {
+			return err
+		}
+		if err := d.os.Remove(cclDst); err != nil {
+			return err
+		}
+
+		if err := d.os.CopyFile(filepath.Join(ossProtobufSrc, relPath), ossDst); err != nil {
```

Comment:
I think `CopyFile` will just overwrite the existing file so you don't have to explicitly remove them first?

Reply:
I wanted to make sure no spare files were left around — deleting a source file shouldn't leave the output file still around in this staging area!

Reply:
Sorry, can you go into more depth about what error case you're specifically trying to address here? If you're just being careful I think simplifying to just do the `CopyFile` instead of `Remove`ing anything ahead of time would be best. If this logic is specifically to address a possible edge case though then at least a comment should be left here because it's a little smelly.

Reply:
Ah, you're right!  Because we're doing individual files for this call and not an entire directory (like we do below for `d.os.CopyAll`), there's no edge-cases involved here. I'll remove the extra `d.os.Remove` on these files 😄

Reply:
makes sense